tests/test_smoke.py

```diff
@@ -325,23 +331,50 @@ def test_env_check():
 # ---------- file_mounts ----------
 def test_file_mounts():
     name = _get_cluster_name()
+    test_commands = [
+        *storage_setup_commands,
+        f'sky launch -y -c {name} examples/using_file_mounts.yaml',
+        f'sky logs {name} 1 --status',  # Ensure the job succeeded.
+    ]
     test = Test(
         'using_file_mounts',
-        [
-            'touch ~/tmpfile',
-            'mkdir -p ~/tmp-workdir',
-            'touch ~/tmp-workdir/tmp\ file',
-            'touch ~/tmp-workdir/foo',
-            'ln -f -s ~/tmp-workdir/ ~/tmp-workdir/circle-link',
-            f'sky launch -y -c {name} examples/using_file_mounts.yaml',
-            f'sky logs {name} 1 --status',  # Ensure the job succeeded.
-        ],
+        test_commands,
         f'sky down -y {name}',
         timeout=20 * 60,  # 20 mins
     )
     run_one_test(test)
```

Comment:
I added cleanup commands in [d7afd8d](https://github.com/skypilot-org/skypilot/pull/1445/commits/d7afd8d27b5db0394985d7bec29698bd140dd349), but that causes transient test failures because a cleanup at the end of a test might delete the files in use by another test. I tried to use `pytest_sessionfinish` [hooks](https://docs.pytest.org/en/latest/reference/reference.html#_pytest.hookspec.pytest_sessionfinish) but that does not work with the parallelization we use in pytest-xdist - each process runs just one test and terminates, so the cleanup code is called n times instead of just once (see [issue](https://github.com/pytest-dev/pytest-xdist/issues/271)). 

To avoid adding more complexity to our testing code, I decided to exclude cleanup for now (since these are tiny files anyway). lmk if you think we should still implement something

Reply:
Sounds good! It seems pretty difficult to implement cleanup. I have another idea if you have time; instead of creating a bunch of files and directories, can we somehow move these commands and use `import tempfile` to generate temporary files and directories on local in that case?